File: src/main/kotlin/coraythan/keyswap/auctions/purchases/PurchaseStats.kt
package coraythan.keyswap.auctions.purchases

import coraythan.keyswap.expansions.Expansion

data class PurchaseStats(
        val standardCount: Long,
        val offerCount: Long,
        val auctionCount: Long,

        val data: List<PurchaseCurrencyData>
)

data class PurchaseCurrencyData(
        val currency: String,
        val expansion: Expansion?,
        val data: List<PurchaseAmountData>
)

data class PurchaseAmountData(
        // Average
        val amount: Int,
        val sas: Int,
        val count: Int
)

File: ui/src/auctions/purchases/PurchaseStats.ts
import { Expansion } from "../../generated-src/Expansion"

export interface PurchaseStats {
    standardCount: number
    offerCount: number
    auctionCount: number

    data: PurchaseCurrencyData[]
}

export interface PurchaseCurrencyData {
    currency: string
    expansion?: Expansion
    data: PurchaseAmountData[]
}

export interface PurchaseAmountData {
    amount: number
    sas: number
    count: number
}
